baseapp/baseapp.go

```diff
@@ -676,6 +681,31 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// is a branch of a branch.
 	runMsgCtx, msCache := app.cacheTxContext(ctx, txBytes)
 
+	// Also emit the following events, so that txs can be indexed by these
+	// indices:
+	// - signature (via `tx.signature='<sig_as_base64>'`),
+	// - address and sequence (via `message.sender=<addr> AND tx.sequence='<seq>' `).
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
```

Comment:
It feels more correct to emit the `tx.*` events in baseapp, however, I didn't find a way to do it without baseapp depending on x/auth. Currently, I import `SigVerifiableTx`, and also copy-pasted `signatureDataToBz` on L803.

We could maybe put the event-emitting in an ante decorator? It would make deps less messy.

Reply:
I would highly highly advocate against having any sort of module dependency in BaseApp. I already see we have a dep on `x/auth/migrations/legacytx` which is pretty unfortunate. Still, I would push for limiting exposure to modules in BaseApp.

I would go with your later approach.

Reply:
I moved event-emitting code to antehandlers.